src/components/Table/Internal/Body/Scroller.tsx

```diff
@@ -0,0 +1,211 @@
+      const { height: stickyHeaderHeight = 0 } =
+        scrollHeaderRef?.current?.getBoundingClientRect?.() || {};
+      const { height: viewportHeight } = document.body.getBoundingClientRect();
+
+      let buttonTop: number = 0;
+
+      if (scrollBodyTop > 0) {
+        // When the top of the table is in the viewport
+
+        if (scrollBodyBottom > viewportHeight) {
+          // When bottom of the table is out of the viewport
+          buttonTop = (viewportHeight - scrollBodyTop) / 2;
+        } else if (scrollBodyBottom < viewportHeight) {
+          // When full table is in the viewport
+          buttonTop = scrollBodyHeight / 2;
+        }
+      } else if (scrollBodyTop < 0) {
+        // When the top of the table is out the viewport
+
+        if (scrollBodyBottom > viewportHeight) {
+          // When bottom of the table is out of the viewport
+          buttonTop = Math.abs(scrollBodyTop) + viewportHeight / 2;
+        } else if (scrollBodyBottom < viewportHeight) {
+          // When bottom of the table is in the viewport
+          buttonTop =
+            Math.abs(scrollBodyTop) +
+            (viewportHeight - (viewportHeight - scrollBodyBottom)) / 2;
+        }
+      }
+      setButtonStyle({
+        top: buttonTop + stickyHeaderHeight - BUTTON_HEIGHT / 2,
+      });
+    }, []);
+
+    const debouncedComputePosition = useDebounce(computePosition, 500);
+
+    const onMouseEnter = useCallback(() => {
+      setVisible(true);
+      computePosition();
+    }, []);
+
+    const onMouseLeave = useCallback(() => setVisible(false), []);
+
+    const onClick = (scrollDirection: 'left' | 'right'): void => {
+      let scrollLeft;
+      if (scrollDirection === 'left') {
+        scrollLeft = scrollOffsets
+          .slice()
+          .reverse()
+          .find(
+            (leftOffset: number) =>
+              leftOffset < scrollBodyRef.current.scrollLeft
+          );
+      } else {
+        scrollLeft = scrollOffsets.find(
+          (leftOffset: number) => leftOffset > scrollBodyRef.current.scrollLeft
+        );
+      }
+      scrollBodyRef.current.scrollTo({
+        left: scrollLeft,
+        behavior: 'smooth',
+      });
+    };
+
+    const onBodyScroll = (): void => {
+      const bodyScrollLeft = scrollBodyRef.current.scrollLeft;
+      const bodyWidth = scrollBodyRef.current.clientWidth;
+      const bodyScrollWidth = scrollBodyRef.current.scrollWidth;
+      if (bodyScrollLeft === 0) {
+        setLeftButtonVisible(false);
+        setRightButtonVisible(true);
+      } else {
+        setLeftButtonVisible(true);
+        if (bodyScrollWidth - bodyScrollLeft - bodyWidth === 0) {
+          setRightButtonVisible(false);
+        } else {
+          setRightButtonVisible(true);
+        }
+      }
+    };
+
+    useImperativeHandle(ref, () => ({
+      onBodyScroll,
+    }));
+
+    useEffect(() => {
+      document.addEventListener('scroll', debouncedComputePosition);
```

Comment:
Maybe scope this event listener to the table ref?

Reply:
this needs to be on the document as the table can go out of view and we need to compute the position on the remaining area visible in the viewport